src/test/ui/incoherent-inherent-impls/needs-has-incoherent-impls.rs

```diff
@@ -0,0 +1,34 @@
+// aux-build:extern-crate.rs
+#![feature(rustc_attrs)]
+extern crate extern_crate;
+
+impl extern_crate::StructWithAttr { //~ ERROR
```

Comment:
```suggestion
impl extern_crate::StructWithAttr { //~ ERROR cannot define inherent `impl` for a type outside of crate
```
Could you use human-readable annotations in all cases?